src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/_params.py

```diff
@@ -58,16 +58,32 @@ def list_app_service_plan_names(prefix, action, parsed_args, **kwargs):#pylint:
 register_cli_argument('webapp deployment slot', 'auto_swap_slot', help='target slot to swap', default='production')
 register_cli_argument('webapp deployment slot', 'disable', help='disable auto swap', action='store_true')
 
-switch_options = ['on', 'off']
-register_cli_argument('webapp log set', 'application_logging', choices=switch_options, type=str.lower, help='configure application logging to file system')
+two_states_switch = [True, False]
+def two_states_switch_type(value=None):
+    return value.lower() == 'true' if value else None
+
+register_cli_argument('webapp log set', 'application_logging', choices=two_states_switch, type=two_states_switch_type, help='configure application logging to file system')
+register_cli_argument('webapp log set', 'detailed_error_messages', choices=two_states_switch, type=two_states_switch_type, help='configure detailed error messages')
+register_cli_argument('webapp log set', 'failed_request_tracing', choices=two_states_switch, type=two_states_switch_type, help='configure failed request tracing')
+register_cli_argument('webapp log set', 'level', choices=['error', 'warning', 'information', 'verbose'], type=str.lower, help='logging level')
 server_log_switch_options = ['off', 'storage', 'filesystem']
 register_cli_argument('webapp log set', 'web_server_logging', choices=server_log_switch_options, type=str.lower, help='configure Web server logging')
-register_cli_argument('webapp log set', 'detailed_error_messages', choices=switch_options, type=str.lower, help='configure detailed error messages')
-register_cli_argument('webapp log set', 'failed_request_tracing', choices=switch_options, type=str.lower, help='configure failed request tracing')
-register_cli_argument('webapp log set', 'level', choices=['error', 'warning', 'information', 'verbose'], type=str.lower, help='logging level')
 
 register_cli_argument('webapp log tail', 'provider', help="scope the live traces to certain providers/folders, for example:'application', 'http' for server log, 'kudu/trace', etc")
 register_cli_argument('webapp log download', 'log_file', default='webapp_logs.zip', help='the downloaded zipped log file path')
 
-register_cli_argument('webapp config appsettings', 'settings', nargs='+', help="space separated app settings in a format of 'name'='value'")
+register_cli_argument('webapp config appsettings', 'settings', nargs='+', help="space separated app settings in a format of <name>=<value>")
 register_cli_argument('webapp config appsettings', 'setting_names', nargs='+', help="space separated app setting names")
```

Comment:
Ah, gotcha, I thought they were for the same command. As far as exposing a generic "settings" parameter, we have avoided this even when Xplat has done this because you don't get tab completion and it's hard to know what you are supposed to or can input (generic update is the big exception to this, but generic update can be very hard to use). When the parameters are exposed as arguments, it's clear what they do. What I did in the case of storage was lump those arguments into an "arg_group" so they are logically grouped together. That would be my suggestion instead of key=value settings.